src/tools/cephfs/top/cephfs-top

```diff
@@ -1191,6 +1191,10 @@ if __name__ == '__main__':
             raise argparse.ArgumentTypeError(
                 'Refresh interval should be greater than or equal to'
                 f' {MIN_REFRESH_INTERVAL}')
+        elif isinstance(x, int):
+            value = x * 10
+        else:
+            value = int(round(x, 1) * 10)
```

Comment:
What's the point of honouring fractional seconds then? You are rounding it to the nearest integer anyway. So why don't we just accept only integer input for `--delay` option, which fixes the issue. curses.halfdelay only accepts integer arguments between 1 and 255. @vshankar Any reason we honour fractional seconds?

I'm attaching the code I've done to fix the same.

[fix.txt](https://github.com/ceph/ceph/files/11109609/fix.txt)

Reply:
> What's the point of honouring fractional seconds then? You are rounding it to the nearest integer anyway. So why don't we just accept only integer input for `--delay` option, which fixes the issue. curses.halfdelay only accepts integer arguments between 1 and 255. @vshankar Any reason we honour fractional seconds?

Nothing specific - just that users can choose 1.5s as delay. top(1) supports that.

> 
> I'm attaching the code I've done to fix the same.
> 
> [fix.txt](https://github.com/ceph/ceph/files/11109609/fix.txt)

Reply:
There won't be a noticeable difference in refresh delay having a fractional second with it's nearest integer. So I think it's better avoid it, if there's no specific need.

Reply:
Understood. Thanks for clarifying.

Reply:
for eg: if user enters delay= 12.43, then `int(round(delay, 1) * 10)` will be `124` and `curses.halfdelay()` accepts 124 and waits for `124/10 = 12.4s` which was expected by the user.

Reply:
Its correct that there's not going to be much difference by having the decimal places in delay, but as discussed by @vshankar to keep the fractional seconds supportability, I went ahead this way.

Reply:
I could have misread the docs and misunderstood that halfdelay accepts fractional values rounded upto one decimal place. I'm fine with integer only values if that's not the case.

Reply:
No, it accepts only integer values. There's no reason to make the user think that halfdelay() processes fractional values.